### PR TITLE
TLSStream Cert Connector Caching

### PR DESCRIPTION
**Pull request by sima-fastly**
Add connector caching to outbound connections

### CODE REVIEW

**Review by deg4uss3r — COMMENTED** [review]
Everything else here looks good, I'd prefer not to store a `String` and use a `&str` and not have to `.clone()` the `connector` but I realize that might lead into a pretty large lifetime mess.

File: src/connmgr/tls.rs [hunk 1]
```diff
@@ -475,6 +475,62 @@ fn apply_wants(e: &ssl::Error, interests: &mut Option<mio::Interest>) {
...
+            cache: Arc::new(Mutex::new(HashMap::new())),
+        }
+    }
+
+    pub fn get_or_create_connector(
+        &self,
+        domain: &str,
+        verify_mode: VerifyMode,
+    ) -> Result<SslConnector, ErrorStack> {
+        let mut cache = self.cache.lock().unwrap();
```
[the remark this answers is not in the document]
Comment (sima-fastly) [reply src/connmgr/tls.rs:507]: Here I have followed the pattern used in this codebase. but I agree with you that better err handling would be beneficial so I am going to use the suggested expect line.

**Comment by sima-fastly** [comment]
> Everything else here looks good, I'd prefer not to store a `String` and use a `&str` and not have to `.clone()` the `connector` but I realize that might lead into a pretty large lifetime mess.

&str is more flexible than String and aligns with rest of the code logic 
your comment about .clone() made me think about memory usage concerns in case of high volume connections so I updated to use Arc to address this.

**Comment by jkarneges** [comment]
Thanks @sima-fastly for starting this. Continuing in #48063

**Pull request closed without merging**